// app_shortcuts/index.js
const electronLocalshortcut = require('electron-localshortcut');
const {globalShortcut} = require('electron');

let win;

var map = {
    'Ctrl+N': nextClicked,
    'Ctrl+P': prevClicked,
    'Space': pauseClicked,
};

var globalMap={
    'MediaNextTrack': nextClicked,
    'MediaPreviousTrack': prevClicked,
    'MediaPlayPause': pauseClicked,
    'Ctrl+3': likeClicked,
    'Ctrl+4': repeatClicked
};


function likeClicked() {
    win.webContents.executeJavaScript("document.getElementsByClassName('playbackSoundBadge__like')[0].click()");
}

function nextClicked() {
    win.webContents.executeJavaScript("document.getElementsByClassName('playControls__next')[0].click()");
}

function prevClicked() {
    win.webContents.executeJavaScript("document.getElementsByClassName('playControls__prev')[0].click()");
}

function pauseClicked() {
    win.webContents.executeJavaScript("document.getElementsByClassName('playControls__play')[0].click()");
}

function repeatClicked(){
    win.webContents.executeJavaScript("document.getElementsByClassName('repeatControl')[0].click()");
}

function initShorts(window) {
    win=window;
    for (let key in map) {
        electronLocalshortcut.register(win, key, () => {
            map[key].call();
        });
    }
    for (let key in globalMap) {
        globalShortcut.register(key,()=>{
            globalMap[key].call();
        })
    }
}

function destShorts(){
    win=null;
    electronLocalshortcut.unregisterAll(win);
}

module.exports={
    initShorts,
    destShorts,
    likeClicked,
    nextClicked,
    prevClicked,
    pauseClicked,
    repeatClicked
};
