// app_shortcuts/index.js
const electronLocalshortcut = require('electron-localshortcut');
const {globalShortcut} = require('electron');
var {ipcMain} = require('electron');

let win;

var map = {
    'Ctrl+N': nextClicked,
    'Ctrl+P': prevClicked,
    'Space': pauseClicked,
};

var globalMap = {
    'MediaNextTrack': nextClicked,
    'MediaPreviousTrack': prevClicked,
    'MediaPlayPause': pauseClicked,
    'Ctrl+3': likeClicked,
    'Ctrl+4': repeatClicked
};

var repeatClickedJScript = `
    var repeatBtn = document.getElementsByClassName('repeatControl')[0];
    repeatBtn.click();
    if(repeatBtn.className.includes("m-none")){
        new Notification('Repeat Mode',{
            body:'Repeat has been disabled',
            silent:true
        });
    }else{
        new Notification('Repeat Mode',{
            body:'Repeat has been enabled',
            silent:true
        });
    }
`;

var likeClicked = `
    var likeBtn = document.getElementsByClassName('playbackSoundBadge__like')[0];
    var title = document.getElementsByClassName('playbackSoundBadge__title')[0].title;
    var imageUrl = document.getElementsByClassName('playbackSoundBadge__avatar')[0].children[0].children[0].style.backgroundImage;
    imageUrl = imageUrl.substring(5,imageUrl.length-2);

    likeBtn.click();
    if(likeBtn.className.includes("sc-button-selected")){
        new Notification('Liked',{
            body:title,
            silent:true,
            icon:imageUrl
        });
    }else{
        new Notification('Disliked',{
            body:title,
            silent:true,
            icon:imageUrl
        });
    }
`;


function likeClicked() {
    win.webContents.executeJavaScript(likeClicked);
}

function nextClicked() {
    win.webContents.executeJavaScript("document.getElementsByClassName('playControls__next')[0].click()");
}

function prevClicked() {
    win.webContents.executeJavaScript("document.getElementsByClassName('playControls__prev')[0].click()");
}

function pauseClicked() {
    win.webContents.executeJavaScript("document.getElementsByClassName('playControls__play')[0].click()");
}

function repeatClicked() {
    win.webContents.executeJavaScript(repeatClickedJScript);
}

function initShorts(window) {
    win = window;
    for (let key in map) {
        electronLocalshortcut.register(win, key, () => {
            map[key].call();
        });
    }
    for (let key in globalMap) {
        globalShortcut.register(key, () => {
            globalMap[key].call();
        })
    }
}

function destShorts() {
    win = null;
    electronLocalshortcut.unregisterAll(win);
}

module.exports = {
    initShorts,
    destShorts,
    likeClicked,
    nextClicked,
    prevClicked,
    pauseClicked,
    repeatClicked
};
