// app_shortcuts/index.js
const electronLocalshortcut = require('electron-localshortcut');

let win;

var map = {
    'L': likeClicked,
    'MediaNextTrack': nextClicked,
    'N': nextClicked,
    'MediaPreviousTrack': prevClicked,
    'P': prevClicked,
    'MediaPlayPause': pauseClicked,
    'Space': pauseClicked,
    'A':repeatClicked
};


function likeClicked() {
    win.webContents.executeJavaScript("document.getElementsByClassName('playbackSoundBadge__like')[0].click()");
}

function nextClicked() {
    win.webContents.executeJavaScript("document.getElementsByClassName('playControls__next')[0].click()");
}

function prevClicked() {
    win.webContents.executeJavaScript("document.getElementsByClassName('playControls__prev')[0].click()");
}

function pauseClicked() {
    win.webContents.executeJavaScript("document.getElementsByClassName('playControls__play')[0].click()");
}

function repeatClicked(){
    win.webContents.executeJavaScript("document.getElementsByClassName('repeatControl')[0].click()");
}

function initShorts(window) {
    win=window;
    for (let key in map) {
        electronLocalshortcut.register(win, key, () => {
            map[key].call(win);
        });
    }
}

function destShorts(){
    win=null;
    electronLocalshortcut.unregisterAll(win);
}

module.exports={
    init:initShorts,
    dest:destShorts
};
